cmake/Modules/FunctionQbtCXXCompilerAndModeCheck.cmake

```diff
@@ -0,0 +1,19 @@
+function(qbt_cxx_compiler_and_mode_check _min_std)
+    # ensure the compiler fully supports the minimum required C++ mode.
+    if(NOT CMAKE_CXX${_min_std}_STANDARD__HAS_FULL_SUPPORT)
+        message(FATAL_ERROR "${PROJECT_NAME} requires a compiler with full C++${_min_std} support")
+    endif()
+
+    # now we know that the compiler fully supports the minimum required C++ mode,
+    # but we must still prevent the user or compiler from forcing/defaulting to an insufficient C++ mode
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD ${_min_std} PARENT_SCOPE)
+    elseif((CMAKE_CXX_STANDARD VERSION_LESS ${_min_std}) OR (CMAKE_CXX_STANDARD VERSION_EQUAL 98))
+        message(FATAL_ERROR "${PROJECT_NAME} has to be built with at least C++${_min_std} mode.")
+    endif()
+
+    set(CMAKE_CXX_STANDARD_REQUIRED ON PARENT_SCOPE)
```

Comment:
Is PARENT_SCOPE required here? The CMAKE_CXX_STANDARD_REQUIRED variable is global already.

Reply:
same as https://github.com/qbittorrent/qBittorrent/pull/12480#discussion_r414786961